modules/behaviors/MapInteractionBehavior.js

```diff
@@ -362,4 +432,55 @@ export class MapInteractionBehavior extends AbstractBehavior {
     }
   }
 
+  /**
+   * _updatePinchState
+   * Updates the pinch state by recalculating the scale change and applying damping if necessary.
+   * This method is called during touch movements to dynamically adjust the map's scale.
+   */
+  _updatePinchState() {
+    const touchPoints = Object.values(this.activeTouches);
+    if (touchPoints.length === 2) {
+        const [first, second] = touchPoints;
+        const currentDistance = Math.hypot(first.x - second.x, first.y - second.y);
+
+        if (this._initialPinchDistance !== null) {
+            const scaleChange = currentDistance / this._initialPinchDistance;
+            const currentZoom = this.context.viewport.transform.zoom;
+
+            // Only apply changes if the scale change is significant
+            if (Math.abs(scaleChange - 1) > 0.01) {  // Adjust this threshold as needed
+                const baseDamping = 0.65;
+                const dampingFactor = currentZoom > 16 ? 0.25 : (currentZoom / 16) * baseDamping + 0.25;
+                const adjustedScaleChange = 1 + (scaleChange - 1) * dampingFactor;
+
+                this._applyPinchZoom(adjustedScaleChange);
+            }
+        }
+        this._initialPinchDistance = currentDistance;
+    } else {
+        this._initialPinchDistance = null;
+    }
+  }
+
+
+  /**
+   * _applyPinchZoom
+   * Applies the calculated scale change to the viewport. This method ensures that the zoom level updates are throttled to prevent excessive updates.
+   * @param {number} scaleChange - The calculated scale change to be applied.
+   */
+  _applyPinchZoom(scaleChange) {
+    const now = performance.now();
+    if (now - this.lastZoomUpdate < 50) return;
+    this.lastZoomUpdate = now;
+
+    const viewport = this.context.viewport;
+    const currentZoom = viewport.transform.zoom;
+    const targetZoom = currentZoom * Math.log2(scaleChange + 1);
+
+    const clampedZoom = Math.max(MIN_Z, Math.min(targetZoom, MAX_Z));
+
+    requestAnimationFrame(() => {
```

Comment:
You won't want to do this in `requestAnimationFrame` callback - the `PixiRenderer.js` code already will defer changes to the map transform, so you can just call `map.zoom(val)` directly.  (By deferring it this way, you can create a race condition where different changes to the transform are fighting with each other).